timemachine/potentials/potential.py

```diff
@@ -48,6 +48,9 @@ def __call__(self, conf: Conf, box: Optional[Box]) -> float | Array:
     def to_gpu(self, precision: Precision) -> "BoundGpuImplWrapper":
         return self.potential.to_gpu(precision).bind(np.asarray(self.params))
 
+    def __post_init__(self):
+        self.params = self.params.astype(np.float64).reshape(self.potential.params_shape())
```

Comment:
Have we verified that this doesn't break jax traceability of code that goes through `BoundPotential`?

I think my personal preference would be to assert on the shape here rather than automagically reshape (since I think it's useful to enforce shape discipline at the call sites), but could be persuaded otherwise.

Reply:
iirc - it's the other way around. Assertions on param shapes can trigger jax traceability exceptions (since they're not traceable).

Reply:
>  Have we verified that this doesn't break jax traceability of code that goes through BoundPotential?

(PS this shouldn't break jax traceability - if `self.params` is a `jax.TracedArray`, the methods `astype` and `reshape` are defined methods on the `jax.TracedArray` object).